src/super_gradients/training/models/optical_flow_models/raft/raft_variants.py

```diff
@@ -0,0 +1,140 @@
+import copy
+from typing import Union, Optional, Callable
+
+import torch
+import torch.nn as nn
+from omegaconf import DictConfig
+
+from super_gradients.common.object_names import Models
+
+from super_gradients.common.registry import register_model
+from super_gradients.module_interfaces import SupportsReplaceInputChannels
+from super_gradients.module_interfaces.exportable_optical_flow import ExportableOpticalFlowModel
+from super_gradients.training.models import get_arch_params, SgModule
+from super_gradients.training.utils.utils import HpmStruct, get_param
+
+from .raft_base import Encoder, ContextEncoder, FlowIterativeBlock
+
+
+class RAFT(ExportableOpticalFlowModel, SgModule):
+    def __init__(self, in_channels, encoder_params, corr_params, flow_params, num_classes):
+        super().__init__()
+
+        self.in_channels = in_channels
+
+        self.feature_encoder = Encoder(
+            in_channels=self.in_channels,
+            in_planes=encoder_params.in_planes,
+            output_dim=encoder_params.fnet.output_dim,
+            norm_fn=encoder_params.fnet.norm_fn,
+            dropout=encoder_params.dropout,
+        )
+
+        self.context_encoder = ContextEncoder(
+            in_channels=self.in_channels,
+            in_planes=encoder_params.in_planes,
+            hidden_dim=encoder_params.hidden_dim,
+            context_dim=encoder_params.context_dim,
+            output_dim=encoder_params.cnet.output_dim,
+            norm_fn=encoder_params.cnet.norm_fn,
+            dropout=encoder_params.dropout,
+        )
+
+        self.flow_iterative_block = FlowIterativeBlock(encoder_params, encoder_params.update_block.hidden_dim, flow_params, corr_params.alternate_corr)
+
+    def freeze_bn(self):
+        for m in self.modules():
+            if isinstance(m, nn.BatchNorm2d):
+                m.eval()
+
+    @staticmethod
+    def coords_grid(batch, ht, wd, device):
+        coords = torch.meshgrid(torch.arange(ht, device=device), torch.arange(wd, device=device))
+        coords = torch.stack(coords[::-1], dim=0).float()
+        return coords[None].repeat(batch, 1, 1, 1)
+
+    def initialize_flow(self, img):
+        """Flow is represented as difference between two coordinate grids flow = coords1 - coords0"""
+        N, C, H, W = img.shape
+        coords0 = self.coords_grid(N, H // 8, W // 8, device=img.device)
+        coords1 = self.coords_grid(N, H // 8, W // 8, device=img.device)
+
+        # optical flow computed as difference: flow = coords1 - coords0
+        return coords0, coords1
+
+    def forward(self, x, **kwargs):
+        """Estimate optical flow between pairs of frames"""
+
+        image1 = x[:, 0]
```

Comment:
1) The way of passing input image pair as [B,2,C,H,W] should be reflected in docstring
2) No kwargs